logstash-core/spec/conditionals_spec.rb

```diff
@@ -424,18 +521,78 @@ def multi_receive(events)
       }
     CONFIG
 
-    sample({"type" => "original"}) do
-      # puts subject.inspect
-      expect(subject[0].get("cond1")).to eq(nil)
+    sample_one({"type" => "original"}) do
+      expect(subject.length).to eq(3)
+      subject.sort! {|a, b| a.get("type") <=> b.get("type")}
+
+      expect(subject[0].get("type")).to eq("clone1")
+      expect(subject[0].get("cond1")).to eq("true")
       expect(subject[0].get("cond2")).to eq(nil)
 
-      expect(subject[1].get("type")).to eq("clone1")
```

Comment:
@andrewvc same answer here :) I handled this via `sort`. `select` was my first attempt, but it requires me to do 2 assertions for each field then (obviously have to make sure each event has a `type` and similar then -> this is much shorter and easier to follow imo since it has no logic in the asserts).